drivers/gpio/gpio_sedi.c

```diff
@@ -0,0 +1,329 @@
+
+	return 0;
+}
+
+static int gpio_sedi_set_masked_raw(const struct device *dev,
+				    uint32_t mask,
+				    uint32_t value)
+{
+	gpio_sedi_write_raw(dev, (mask & value), false);
+
+	return 0;
+}
+
+static int gpio_sedi_set_bits_raw(const struct device *dev, uint32_t pins)
+{
+	gpio_sedi_write_raw(dev, pins, false);
+
+	return 0;
+}
+
+static int gpio_sedi_clear_bits_raw(const struct device *dev, uint32_t pins)
+{
+	gpio_sedi_write_raw(dev, pins, true);
+
+	return 0;
+}
+
+static int gpio_sedi_toggle_bits(const struct device *dev, uint32_t pins)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+	uint8_t i;
+
+	for (i = 0; i < config->pin_nums; i++) {
+		if (pins & 0x1) {
+			sedi_gpio_toggle_pin(gpio_dev, i);
+		}
+		pins >>= 1;
+		if (pins == 0) {
+			break;
+		}
+	}
+
+	return 0;
+}
+
+static int gpio_sedi_interrupt_configure(const struct device *dev,
+					 gpio_pin_t pin,
+					 enum gpio_int_mode mode,
+					 enum gpio_int_trig trig)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+	sedi_gpio_pin_config_t pin_config = { 0 };
+
+	/* Only input needs interrupt enabled */
+	pin_config.direction = SEDI_GPIO_DIR_MODE_INPUT;
+	pin_config.enable_wakeup = true;
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		pin_config.enable_interrupt = false;
+	} else {
+		pin_config.enable_interrupt = true;
+		if (mode == GPIO_INT_MODE_LEVEL) {
+			return -1;
+		}
+		switch (trig) {
+		case GPIO_INT_TRIG_LOW:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_FALLING_EDGE;
+			break;
+		case GPIO_INT_TRIG_HIGH:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_RISING_EDGE;
+			break;
+		case GPIO_INT_TRIG_BOTH:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_BOTH_EDGE;
+			break;
+		}
+	}
+	/* Configure interrupt mode */
+	sedi_gpio_config_pin(gpio_dev, pin, pin_config);
+
+	return 0;
+}
+
+static int gpio_sedi_manage_callback(const struct device *dev,
+				     struct gpio_callback *callback,
+				     bool set)
+{
+	struct gpio_sedi_data *data = dev->data;
+
+	gpio_manage_callback(&(data->callbacks), callback, set);
+
+	return 0;
+}
+
+static uint32_t gpio_sedi_get_pending(const struct device *dev)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+
+	return sedi_gpio_get_gisr(gpio_dev, 0);
+}
+
+static const struct gpio_driver_api gpio_sedi_driver_api = {
+	.pin_configure = gpio_sedi_configure,
+	.port_get_raw = gpio_sedi_get_raw,
+	.port_set_masked_raw = gpio_sedi_set_masked_raw,
+	.port_set_bits_raw = gpio_sedi_set_bits_raw,
+	.port_clear_bits_raw = gpio_sedi_clear_bits_raw,
+	.port_toggle_bits = gpio_sedi_toggle_bits,
+	.pin_interrupt_configure = gpio_sedi_interrupt_configure,
+	.manage_callback = gpio_sedi_manage_callback,
+	.get_pending_int = gpio_sedi_get_pending
+};
+
+extern void gpio_isr(IN sedi_gpio_t gpio_device);
+
+static int gpio_sedi_init(const struct device *dev)
+{
```

Comment:
call DEVICE_MMIO_MAP() for register memory remap for paging, see i2c_sedi.c.

Reply:
done